NEWS.md

```diff
@@ -87,6 +87,8 @@ Standard library changes
 
 #### LinearAlgebra
 
+* `rank` can now take a QRPivoted matrix ([#53214]) to allow rank computation on QR-factorized matrices.
```

Comment:
```suggestion
* `rank` can now take a `QRPivoted` matrix to allow rank estimation via QR factorization ([#53214]).
```

Reply:
Done, LMK whether I should merge after this